b2sdk/raw_simulator.py

```diff
@@ -654,6 +656,10 @@ def __init__(self):
         # Set of auth tokens that have expired
         self.expired_auth_tokens = set()
 
+        # Map from auth token to a lock that upload procedure acquires
+        # when utilizing the token
+        self.currently_used_auth_tokens = collections.defaultdict(threading.Lock)
```

Comment:
Is there a danger that this mapping will grow unbounded?  If the the usage pattern creates a new upload token for each upload, every upload would add to the map.  When uploading millions of files it could get big.

Would it be sufficient just to keep a set of the currently used tokens?  (With locking for thread safety, of course.)

Reply:
This change is just for the raw_simulator, which is only used for unit tests in place of the actual cloud service. The problem here is that user reported a bug that clearly shows the CLI did not prevent token re-usage. Before we will try to implement an exposing test, we want to make the simulated cloud service raise alarm whenever someone is trying to reuse the token.

This whole simulator keeps all files in memory etc. I wouldn't worry about growth of the mapping in mentioned line - file content will consume all the memory first.

Reply:
Ah.  Right.  Sounds good then.